services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -98,5 +114,64 @@ def display_name
         errors.add(:prompts, "prompt #{prompt.id} already has a plagiarism rule") if prompt.rules.where(rule_type: TYPE_PLAGIARISM).first&.id
       end
     end
+
+    def log_creation(user_id)
+      if regex?
```

Comment:
Not sure if this is actually be better, but this could be a set of early returns:
```ruby
return send_change_log(...) if ...
return send_change_log(...) if ...
```
And so on.